pandas/core/arrays/categorical.py

```diff
@@ -1554,6 +1560,15 @@ def sort_values(self, inplace=False, ascending=True, na_position="last"):
         na_position : {'first', 'last'} (optional, default='last')
             'first' puts NaNs at the beginning
             'last' puts NaNs at the end
+        key : callable, optional
+            Apply the key function to the values before sorting.
```

Comment:
you might want to give a sample of the key function here

Reply:
Thinking more about this, I'm inclined to remove the key function here. I can't think of a use case for `Categorical` because it supports basically no vectorized operations and `cat.map` doesn't work well with key sorting because it just transforms the `codes`.